lib/packwerk/configuration.rb

```diff
@@ -61,10 +61,12 @@ def all_application_autoload_paths
       bundle_path_match = Bundler.bundle_path.join("**").to_s
 
       all_paths = all_paths.each_with_object([]) do |path_string, paths|
-        # ignore paths inside gems
         path = Pathname.new(path_string)
 
-        next unless path.exist?
+        # ignore paths outside of the Rails root
+        next unless path.exist? && path.realpath.fnmatch(Rails.root.join("**").to_s)
```

Comment:
https://github.com/Shopify/packwerk/pull/21/files#diff-e43184d1a960a3c3821da21b3eb72c83379c2dcc996f6bcfac9aadd6898cd374L64

Reply:
Nitpick: it feels a bit clearer to me if these two conditionals exist on their own lines:
```ruby
next unless path.exist?
next unless path.realpath.fnmatch(Rails.root.join("**").to_s)
```
`path.exist?` is a precondition for the rest of the loop, not just for your new Rails path condition.

Reply:
Can we test this new condition in configuration_test?  Maybe by appending an external path to `Rails.application.config.autoload_paths` beforehand?  Don't forget to restore the paths to their old value.

Reply:
> Can we test this new condition in configuration_test? Maybe by appending an external path to Rails.application.config.autoload_paths beforehand? Don't forget to restore the paths to their old value.

Tested in https://github.com/Shopify/packwerk/pull/68 where this logic was moved into its own class.